analyzers/src/SonarAnalyzer.CSharp/Rules/Hotspots/DoNotHardcodeCredentials.cs

```diff
@@ -146,5 +160,68 @@ private static bool ShouldHandle(SyntaxNode method, SyntaxNode current)
+                return GetStringConstant(left) is { } leftString
+                    && GetStringConstant(syntaxNode.Right) is { } rightString
+                    ? leftString + rightString
+                    : null;
+
+                string GetStringConstant(SyntaxNode node) =>
+                    node.GetStringValue() ?? semanticModel.GetConstantValue(node).Value as string;
+            }
+
+            protected override string GetVariableName(BinaryExpressionSyntax syntaxNode) => null;
+
+            protected override bool ShouldHandle(BinaryExpressionSyntax syntaxNode, SemanticModel semanticModel) => true;
+        }
+
+        private class InterpolatedStringBannedWordsFinder : CredentialWordsFinderBase<InterpolatedStringExpressionSyntax>
+        {
+            public InterpolatedStringBannedWordsFinder(DoNotHardcodeCredentials analyzer) : base(analyzer) { }
+
+            protected override string GetAssignedValue(InterpolatedStringExpressionSyntax syntaxNode, SemanticModel semanticModel) =>
+                syntaxNode.Contents.JoinStr(null, x => x switch
+                {
+                    InterpolationSyntax interpolation => semanticModel.GetConstantValue(interpolation.Expression).Value as string,
+                    InterpolatedStringTextSyntax text => text.TextToken.ToString(),
+                    _ => null
+                } ?? CredentialSeparator.ToString()); // Unknown elements resolved to separator to terminate the keyword-value sequence
+
+            protected override string GetVariableName(InterpolatedStringExpressionSyntax syntaxNode) => null;
+
+            protected override bool ShouldHandle(InterpolatedStringExpressionSyntax syntaxNode, SemanticModel semanticModel) => true;
+        }
+
+        private class InvocationBannedWordsFinder : CredentialWordsFinderBase<InvocationExpressionSyntax>
+        {
+            public InvocationBannedWordsFinder(DoNotHardcodeCredentials analyzer) : base(analyzer) { }
+
+            protected override string GetAssignedValue(InvocationExpressionSyntax syntaxNode, SemanticModel semanticModel)
+            {
+                var allArgs = syntaxNode.ArgumentList.Arguments.Select(x => semanticModel.GetConstantValue(x.Expression).Value as string ?? CredentialSeparator.ToString());
+                try
+                {
+                    return string.Format(allArgs.First(), allArgs.Skip(1).ToArray());
+                }
+                catch (FormatException)
+                {
+                    return null;
+                }
+            }
+
+            protected override string GetVariableName(InvocationExpressionSyntax syntaxNode) => null;
+
+            protected override bool ShouldHandle(InvocationExpressionSyntax syntaxNode, SemanticModel semanticModel) =>
+                syntaxNode.IsMethodInvocation(KnownType.System_String, "Format", semanticModel)
```

Comment:
Maybe it would be useful to create a helper extension method IsMethodInvocationWithParams. For me seems like something that could be useful in other places too, and that way we would not do semanticModel.GetSymbolInfo(syntaxNode).Symbol is IMethodSymbol symbol twice.

Reply:
I checked the other places and no one is checking the args (yet). So maybe later. It's closed closely to each other, so Roslyn internal symbol cache should deal with the performance penalty in this case.